src/plot_api/subroutines.js

```diff
@@ -513,6 +513,13 @@ exports.doModeBar = function(gd) {
         subplotObj.updateFx(fullLayout);
     }
 
+    subplotIds = Plots.getSubplotIds(fullLayout, 'geo');
+    for(i = 0; i < subplotIds.length; i++) {
+        subplotLayout = fullLayout[subplotIds[i]];
+        subplotObj = subplotLayout._subplot;
+        subplotObj.updateFx(fullLayout, subplotLayout);
+    }
```

Comment:
This section is getting a bit overgrown... can we pull these out into the plot modules like:
```js
for(i=0; i < fullLayout._basePlotModules.length; i++) {
    var updateFx = fullLayout._basePlotModules[i].updateFx;
    if(updateFx) updateFx(fullLayout);
}
```

Reply:
Good call. I'll need to make the `updateFx` API consistent across subplots, but yeah, it'll be very much worthwhile.

Reply:
done in https://github.com/plotly/plotly.js/pull/2030/commits/2488f4b71ff79ef00a476f845f7e29fac8b7cc0b